src/Microsoft.Azure.EventHubs/EventData.cs

```diff
@@ -168,12 +152,10 @@ public string Offset
                     object value;
                     if (this.TryGetValue(ClientConstants.OffsetName, out value))
                     {
-                        return (string)value;
-                    }
-                    else
-                    {
-                        throw new ArgumentException(Resources.MissingSystemProperty.FormatForUser(ClientConstants.OffsetName));
+                        return value.ToString();
```

Comment:
>return value.ToString(); [](start = 24, length = 24)

You should not convert a cast to ToString. Cast is expected fail if this is not a string type, whereas ToString() won't. Better to keep the cast as is.

Reply:
Ok, we can use `is` instead to avoid a value cast operation but I'm not aware of the perfomance difference between `is` and excplity cast from `object` to `string`.  

I will leave like before then if the idea is break excplity

Reply:
Let's keep it simple by letting it throw if cast fails. Rest of the changes looked OK.

---
In reply to: [233270548](https://github.com/Azure/azure-event-hubs-dotnet/pull/350#discussion_r233270548) [](ancestors = 233270548)

Reply:
fair enough :)